tools/gulp/util/package-build.ts

```diff
@@ -0,0 +1,111 @@
+  let packagePath = join(DIST_ROOT, 'packages', packageName);
+  let releasePath = join(DIST_ROOT, 'releases', packageName);
+
+  inlineMetadata(packagePath);
+
+  copyFiles(packagePath, '**/*.+(d.ts|metadata.json)', join(releasePath, 'typings'));
+  copyFiles(DIST_BUNDLES, `${packageName}.umd?(.min).js`, join(releasePath, 'bundles'));
+  copyFiles(DIST_BUNDLES, `${packageName}?(.es5).js`, join(releasePath, '@angular'));
+  copyFiles(PROJECT_ROOT, 'LICENSE', releasePath);
+  copyFiles(SOURCE_ROOT, 'README', releasePath);
+  copyFiles(sourcePath, 'package.json', releasePath);
+
+  createTypingFile(releasePath, packageName);
+  createMetadataFile(releasePath, packageName);
+}
+
+/** Builds a module entry-point. If no entry name is specified it builds the whole library. */
+export async function buildModuleEntry(entryFile: string, entryName = 'material') {
+  let moduleName = entryName ? `ng.material.${entryName}` : 'ng.material';
+
+  // List of paths for the specified entrypoint.
+  let fesm2015File = join(DIST_BUNDLES, `${entryName}.js`);
+  let fesm2014File = join(DIST_BUNDLES, `${entryName}.es5.js`);
+  let umdFile = join(DIST_BUNDLES, `${entryName}.umd.js`);
+  let umdMinFile = join(DIST_BUNDLES, `${entryName}.umd.min.js`);
+
+  // Build FESM-2015 bundle file.
+  await createRollupBundle({
+    moduleName: moduleName,
+    entry: entryFile,
+    dest: fesm2015File,
+    format: 'es',
+  });
+
+  // Downlevel FESM-2015 file to ES5.
+  transpileFile(fesm2015File, fesm2014File, {
+    target: ScriptTarget.ES5,
+    module: ModuleKind.ES2015,
+    allowJs: true
+  });
+
+  // Create UMD bundle of FESM-2014 output.
+  await createRollupBundle({
+    moduleName: moduleName,
+    entry: fesm2014File,
+    dest: umdFile,
+    format: 'umd'
+  });
+
+  // Output a minified version of the UMD bundle
+  writeFileSync(umdMinFile, uglify.minify(umdFile, { preserveComments: 'license' }).code);
+}
+
+function copyFiles(fromPath: string, fileGlob: string, outDir: string) {
+  glob(fileGlob, {cwd: fromPath}).forEach(filePath => {
+    let fileDestPath = join(outDir, filePath);
+    mkdirpSync(dirname(fileDestPath));
+    copySync(join(fromPath, filePath), fileDestPath);
+  });
+}
+
+/** Create a typing file that links to the bundled definitions of NGC. */
+function createTypingFile(outputDir: string, entryName: string) {
+  writeFileSync(join(outputDir, `${entryName}.d.ts`),
+    LICENSE_BANNER + '\nexport * from "./typings/index";'
+  );
+}
+
+/** Creates a metadata file that re-exports the metadata bundle inside of the typings. */
+function createMetadataFile(packageDir: string, packageName: string) {
+  const metadataReExport =
+      `{"__symbolic":"module","version":3,"metadata":{},"exports":[{"from":"./typings/index"}]}`;
+  writeFileSync(join(packageDir, `${packageName}.metadata.json`), metadataReExport, 'utf-8');
+}
+
+function inlineMetadata(packagePath: string) {
```

Comment:
`inlineMetadataResources` is another method we import from the `utils`. Therefore I used a different name here. Adding a comment anyways